### PR TITLE
24552 - Update Resolution Section in Outputs + NOA

### PR DESCRIPTION
**Pull request by ArwenQin**
*Issue #:* /bcgov/entity#24552
*Description of changes:*
- Update the Resolution Section wording in NOA
- Update the Resolution Section wording and resolution date format in Alteration/Correction/Amalgamation/Transition outputs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by ArwenQin** [comment]
Updated Correction Output:
image

**Comment by ArwenQin** [comment]
Updated Alteration Output:
image

**Comment by ArwenQin** [comment]
Updated NOA:
image

**Comment by ArwenQin** [comment]
Seems this resolution date section is not in amalgamation/transition filing UI yet, so I can't generate an output to see it. Could you please suggest? Thanks a lot.

File: legal-api/src/legal_api/resources/v2/business/colin_sync.py [hunk 1]
```diff
@@ -289,6 +290,9 @@ def _set_shares(primary_or_holding_business, amalgamation_filing, transaction_id
     amalgamation_filing['shareStructure'] = {'shareClasses': share_classes}
     business_dates = [item.resolution_date.isoformat() for item in primary_or_holding_business.resolutions]
     if business_dates:
+        formatted_dates = [
+            datetime.strptime(date, '%Y-%m-%d').strftime('%B %-d, %Y').replace(' 0', ' ') for date in business_dates
+        ]
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/business/colin_sync.py:295]: No need to format here. This is for colin sync

File: legal-api/src/legal_api/reports/report.py [hunk 2]
```diff
@@ -488,6 +488,11 @@ def _format_transition_data(self, filing):
             self._format_address(filing['offices']['recordsOffice']['mailingAddress'])
         if filing.get('shareStructure', {}).get('shareClasses', None):
             filing['shareClasses'] = filing['shareStructure']['shareClasses']
+            dates = filing['shareStructure'].get('resolutionDates', [])
+            formatted_dates = [
+                datetime.strptime(date, '%Y-%m-%d').strftime('%B %d, %Y').replace(' 0', ' ') for date in dates
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/reports/report.py:493]:
```suggestion
                datetime.fromisoformat(date).strftime(OUTPUT_DATE_FORMAT) for date in dates
```

File: legal-api/src/legal_api/reports/report.py [hunk 3]
```diff
@@ -672,7 +677,11 @@ def _format_alteration_data(self, filing):
                 self._filing.transaction_id, self._business.id)
         if filing['alteration'].get('shareStructure', None):
             filing['shareClasses'] = filing['alteration']['shareStructure'].get('shareClasses', [])
-            filing['resolutions'] = filing['alteration']['shareStructure'].get('resolutionDates', [])
+            dates = filing['alteration']['shareStructure'].get('resolutionDates', [])
+            formatted_dates = [
+                datetime.strptime(date, '%Y-%m-%d').strftime('%B %d, %Y').replace(' 0', ' ') for date in dates
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/reports/report.py:682]:
```suggestion
                datetime.fromisoformat(date).strftime(OUTPUT_DATE_FORMAT) for date in dates
```

File: legal-api/src/legal_api/reports/report.py [hunk 4]
```diff
@@ -1137,7 +1146,11 @@ def _format_party_data(self, filing, prev_completed_filing: Filing):
 
     def _format_share_class_data(self, filing, prev_completed_filing: Filing):  # pylint: disable=too-many-locals; # noqa: E501;
         filing['shareClasses'] = filing.get('correction').get('shareStructure', {}).get('shareClasses')
-        filing['resolutions'] = filing.get('correction').get('shareStructure', {}).get('resolutionDates', [])
+        dates = filing['correction']['shareStructure'].get('resolutionDates', [])
+        formatted_dates = [
+            datetime.strptime(date, '%Y-%m-%d').strftime('%B %d, %Y').replace(' 0', ' ') for date in dates
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/reports/report.py:1151]:
```suggestion
                datetime.fromisoformat(date).strftime(OUTPUT_DATE_FORMAT) for date in dates
```

File: legal-api/src/legal_api/reports/report.py (hunk 3 above)
Comment (ArwenQin) [reply legal-api/src/legal_api/reports/report.py]: Thank you vysakh, updated

File: legal-api/src/legal_api/resources/v2/business/colin_sync.py (hunk 1 above)
Comment (ArwenQin) [reply legal-api/src/legal_api/resources/v2/business/colin_sync.py]: Sure, just restored

**Comment by vysakh-menon-aot** [comment]
> Seems this resolution date section is not in amalgamation/transition filing UI yet, so I can't generate an output to see it. Could you please suggest? Thanks a lot.

- Amalgamation already show the correct format (https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/reports/report.py#L832)
- Transition does not format resolution date, you can add it in `_format_transition_data` 
-

**Comment by ArwenQin** [comment]
> * Amalgamation already show the correct format (https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/reports/report.py#L832)
> * Transition does not format resolution date, you can add it in `_format_transition_data`

Thanks very much. For Transition, already added in `_format_transition_data` (line 492).

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**